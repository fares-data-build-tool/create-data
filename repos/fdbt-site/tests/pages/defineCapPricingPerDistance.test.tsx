import * as React from 'react';
import { shallow } from 'enzyme';
import DefineCapPricingPerDistance from '../../src/pages/defineCapPricingPerDistance';

describe('pages', () => {
    describe('defineCapPricingPerDistance', () => {
        it('should render correctly', () => {
            const tree = shallow(
                <DefineCapPricingPerDistance
                    errors={[]}
                    csrfToken=""
                    capPricePerDistances={{ distanceFrom0: '0' }}
                    numberOfCapInitial={1}
                />,
            );
            expect(tree).toMatchSnapshot();
        });

        it('should render error messaging when errors are passed', () => {
            const tree = shallow(
                <DefineCapPricingPerDistance
                    errors={[
                        {
                            id: `distance-to-0`,
                            errorMessage: 'Distance to is required and needs to be number',
                        },
                    ]}
                    csrfToken=""
                    capPricePerDistances={{
                        distanceFrom0: '0',
                        distanceFrom1: '3',
                        distanceTo0: '',
                        maximumPrice: '4',
                        minimumPrice: '3',
                        pricePerKm1: '5',
                        pricePerKm0: '5',
                    }}
                    numberOfCapInitial={2}
                />,
            );
            expect(tree).toMatchSnapshot();
        });
    });
});
