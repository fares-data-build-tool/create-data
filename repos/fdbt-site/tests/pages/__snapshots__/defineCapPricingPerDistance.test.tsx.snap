// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages defineCapPricingPerDistance should render correctly 1`] = `
<FullColumnLayout
  description="Cap Pricing Per Distance entry page of the Create Fares Data Service"
  errors={Array []}
  title="Cap Pricing Per Distance - Create Fares Data Service"
>
  <CsrfForm
    action="/api/capPricingPerDistance"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <h1
      className="govuk-heading-l"
      id="cap-pricing-per-distance-heading"
    >
      Enter your distance cap details
    </h1>
    <div
      className="flex-container"
    >
      <div
        className="govuk-!-margin-left-1 govuk-!-margin-right-2"
      >
        <FormGroupWrapper
          errorIds={
            Array [
              "cap-pricing-per-distance-minimum-price",
            ]
          }
          errors={Array []}
          hideErrorBar={true}
        >
          <label
            className="govuk-label"
            htmlFor="minimum-price"
          >
            Minimum price(£)
          </label>
          <span
            className="govuk-hint"
            id="minimum-price-hint"
          >
            e.g. £2.99
          </span>
          <div
            className="govuk-input__wrapper"
          >
            <div
              aria-hidden="true"
              className="govuk-input__prefix"
            >
              £
            </div>
            <input
              className="govuk-input govuk-input--width-5"
              id="minimum-price"
              name="minimumPrice"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </FormGroupWrapper>
      </div>
      <div
        className="govuk-!-margin-left-2 govuk-!-margin-right-2"
      >
        <FormGroupWrapper
          errorIds={
            Array [
              "cap-pricing-per-distance-maximum-price",
            ]
          }
          errors={Array []}
          hideErrorBar={true}
        >
          <label
            className="govuk-label"
            htmlFor="maximum-price"
          >
            Maximum price(£)
          </label>
          <span
            className="govuk-hint"
            id="maximum-price-hint"
          >
            e.g. £2.99
          </span>
          <div
            className="govuk-input__wrapper"
          >
            <div
              aria-hidden="true"
              className="govuk-input__prefix"
            >
              £
            </div>
            <input
              className="govuk-input govuk-input--width-5"
              id="maximum-price"
              name="maximumPrice"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </FormGroupWrapper>
      </div>
    </div>
    <div
      className="govuk-grid-row"
    >
      <DistanceRow
        capPricingPerDistanceData={
          Object {
            "distanceFrom0": "0",
            "distanceTo0": "Max",
          }
        }
        errors={Array []}
        numberOfCapToDisplay={1}
        setCapPricingPerDistanceData={[Function]}
      />
      <div
        className="flex-container"
      >
        <button
          className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3 time-restrictions-button-placement"
          id="add-another-button"
          onClick={[Function]}
          type="button"
        >
          Add another row
        </button>
      </div>
      <input
        className="govuk-button govuk-!-margin-left-3"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</FullColumnLayout>
`;

exports[`pages defineCapPricingPerDistance should render error messaging when errors are passed 1`] = `
<FullColumnLayout
  description="Cap Pricing Per Distance entry page of the Create Fares Data Service"
  errors={
    Array [
      Object {
        "errorMessage": "Distance to is required and needs to be number",
        "id": "distance-to-0",
      },
    ]
  }
  title="Cap Pricing Per Distance - Create Fares Data Service"
>
  <CsrfForm
    action="/api/capPricingPerDistance"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={
        Array [
          Object {
            "errorMessage": "Distance to is required and needs to be number",
            "id": "distance-to-0",
          },
        ]
      }
    />
    <h1
      className="govuk-heading-l"
      id="cap-pricing-per-distance-heading"
    >
      Enter your distance cap details
    </h1>
    <div
      className="flex-container"
    >
      <div
        className="govuk-!-margin-left-1 govuk-!-margin-right-2"
      >
        <FormGroupWrapper
          errorIds={
            Array [
              "cap-pricing-per-distance-minimum-price",
            ]
          }
          errors={
            Array [
              Object {
                "errorMessage": "Distance to is required and needs to be number",
                "id": "distance-to-0",
              },
            ]
          }
          hideErrorBar={true}
        >
          <label
            className="govuk-label"
            htmlFor="minimum-price"
          >
            Minimum price(£)
          </label>
          <span
            className="govuk-hint"
            id="minimum-price-hint"
          >
            e.g. £2.99
          </span>
          <div
            className="govuk-input__wrapper"
          >
            <div
              aria-hidden="true"
              className="govuk-input__prefix"
            >
              £
            </div>
            <input
              className="govuk-input govuk-input--width-5"
              id="minimum-price"
              name="minimumPrice"
              onChange={[Function]}
              type="text"
              value="3"
            />
          </div>
        </FormGroupWrapper>
      </div>
      <div
        className="govuk-!-margin-left-2 govuk-!-margin-right-2"
      >
        <FormGroupWrapper
          errorIds={
            Array [
              "cap-pricing-per-distance-maximum-price",
            ]
          }
          errors={
            Array [
              Object {
                "errorMessage": "Distance to is required and needs to be number",
                "id": "distance-to-0",
              },
            ]
          }
          hideErrorBar={true}
        >
          <label
            className="govuk-label"
            htmlFor="maximum-price"
          >
            Maximum price(£)
          </label>
          <span
            className="govuk-hint"
            id="maximum-price-hint"
          >
            e.g. £2.99
          </span>
          <div
            className="govuk-input__wrapper"
          >
            <div
              aria-hidden="true"
              className="govuk-input__prefix"
            >
              £
            </div>
            <input
              className="govuk-input govuk-input--width-5"
              id="maximum-price"
              name="maximumPrice"
              onChange={[Function]}
              type="text"
              value="4"
            />
          </div>
        </FormGroupWrapper>
      </div>
    </div>
    <div
      className="govuk-grid-row"
    >
      <DistanceRow
        capPricingPerDistanceData={
          Object {
            "distanceFrom0": "0",
            "distanceFrom1": "3",
            "distanceTo0": "",
            "distanceTo1": "Max",
            "maximumPrice": "4",
            "minimumPrice": "3",
            "pricePerKm0": "5",
            "pricePerKm1": "5",
          }
        }
        errors={
          Array [
            Object {
              "errorMessage": "Distance to is required and needs to be number",
              "id": "distance-to-0",
            },
          ]
        }
        numberOfCapToDisplay={2}
        setCapPricingPerDistanceData={[Function]}
      />
      <div
        className="flex-container"
      >
        <button
          className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3 time-restrictions-button-placement"
          id="add-another-button"
          onClick={[Function]}
          type="button"
        >
          Add another row
        </button>
        <button
          className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3"
          id="remove-button"
          onClick={[Function]}
          type="button"
        >
          Remove last row
        </button>
      </div>
      <input
        className="govuk-button govuk-!-margin-left-3"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</FullColumnLayout>
`;
